docs/setup-cluster/k8s/internal-task-gateway.rst

```diff
@@ -0,0 +1,128 @@
+.. _internal-task-gateway:
+
+#######################
+ Internal Task Gateway
+#######################
+
+`K8s Gateway APIs <https://gateway-api.sigs.k8s.io/>`_ allow us to expose otherwise internal
+Determined jobs running on remote K8s clusters to Determined master and proxies. This is useful for
+multi-resource manager setups.
+
+The overall setup includes installing and configuring a Gateway controller in the K8s cluster
+external to Determined and configuring the Determined master to use the Gateway controller. Please
+refer to the sections below to see the configuration changes and controller requirements.
+
+.. warning::
+
+   This feature exposes Determined tasks to the outside world. Please ensure that you have the
+   necessary security measures in place to both limit access to the exposed tasks, and secure the
+   communication between the external cluster and the main one. This could include setting up a
+   firewall, using a VPN, IP white-listing, K8s Network Policies, native cloud solutions, or other
+   security measures. Some Determined tasks including JupyterLab Notebooks and shells already have
+   secure transport built-in. Tensorboards do not use TLS currently so a tunneling solution should
+   be considered.
```

Comment:
Enabling this feature exposes Determined tasks to the outside world. It is crucial to implement appropriate security measures to restrict access to exposed tasks and secure communication between the external cluster and the main cluster. Recommended measures include:

- Setting up a firewall
- Using a VPN
- Implementing IP whitelisting
- Configuring Kubernetes Network Policies
- Employing other security measures as needed

Some tasks including JupyterLab notebooks and shells already have secure transport built-in. 
Since TensorBoards do not use TLS, you should consider a tunneling solution.